src/main/java/com/mparticle/kits/PilgrimSdkKit.java

```diff
@@ -0,0 +1,191 @@
+     * PilgrimUserInfo instance.
+     */
+    private PilgrimUserInfo getUserInfo() {
+        PilgrimUserInfo info = PilgrimSdk.get().getUserInfo();
+        // can be null if it hasn't been set previously
+        if (info == null) {
+            info = new PilgrimUserInfo();
+        }
+        return info;
+    }
+
+    /* Section: User attribute set */
+    @Override
+    public void onSetAllUserAttributes(Map<String, String> attributes, Map<String, List<String>> attributeLists, FilteredMParticleUser filteredMParticleUser) {
+        // NOTE: attributeList not supported
+        // Is this ID correct?
+        updateUser(filteredMParticleUser.mpUser);
+        PilgrimUserInfo info = getUserInfo();
+        for (Map.Entry<String, String> entry : attributes.entrySet()) {
+            info.put(entry.getKey(), entry.getValue());
+        }
+        PilgrimSdk.get().setUserInfo(info);
+    }
+
+
+    @Override
+    public void onIncrementUserAttribute(String s, String s1, FilteredMParticleUser filteredMParticleUser) {
+        // Ignored,  Not supported atm
+    }
+
+    @Override
+    public void onRemoveUserAttribute(String s, FilteredMParticleUser filteredMParticleUser) {
+        PilgrimUserInfo info = getUserInfo();
+        // Do we want to log if it doesn't exist?
+        info.remove(s);
+    }
+
+    @Override
+    public void onSetUserAttribute(String s, Object o, FilteredMParticleUser filteredMParticleUser) {
+        PilgrimUserInfo info = getUserInfo();
+        info.put(s, o.toString());
+        // update
+        PilgrimSdk.get().setUserInfo(info);
+    }
+
+    @Override
+    public void onSetUserTag(String s, FilteredMParticleUser filteredMParticleUser) {
+        // Ignored, not supported atm.
+    }
+
+    @Override
+    public boolean supportsAttributeLists() {
+        // We don't support attribute lists
+        // the PilgrimSdk currently only
+        // allows single k => value attributes.
+        return false;
+    }
+
+    @Override
+    public void onSetUserAttributeList(String s, List<String> list, FilteredMParticleUser filteredMParticleUser) {
+        // Ignored, not supported
+    }
+
+    @Override
+    public void onConsentStateUpdated(ConsentState consentState, ConsentState consentState1, FilteredMParticleUser filteredMParticleUser) {
+        // Ignore?
+    }
+
+    @Override
+    public void onIdentifyCompleted(MParticleUser mParticleUser, FilteredIdentityApiRequest filteredIdentityApiRequest) {
+        updateUser(mParticleUser);
+    }
+
+    @Override
+    public void onLoginCompleted(MParticleUser mParticleUser, FilteredIdentityApiRequest filteredIdentityApiRequest) {
+        updateUser(mParticleUser);
+    }
+
+    @Override
+    public void onModifyCompleted(MParticleUser mParticleUser, FilteredIdentityApiRequest filteredIdentityApiRequest) {
+        updateUser(mParticleUser);
+    }
+
+    private void updateUser(MParticleUser mParticleUser) {
+        PilgrimUserInfo info = getUserInfo();
```

Comment:
ok, so i think this is my last comment in the PR. Based on their responses, I think we should be using the actual customer id here (maybe like you originally had it?) BUT also passing this mparticle user id in, with a key like "mParticleUserId", that way our server can do the right thing with it based on their recommendations later on.

Reply:
ohhh ok, that sounds much better.
I totally misunderstood that from the previous conversation . 

Will update accordingly.

Reply:
Updated